pkg/service/backup/model.go

```diff
@@ -39,21 +39,34 @@ type ListItem struct {
 
 // Target specifies what should be backed up and where.
 type Target struct {
-	Units            []Unit            `json:"units,omitempty"`
-	DC               []string          `json:"dc,omitempty"`
-	Location         []Location        `json:"location"`
-	Retention        int               `json:"retention"`
-	RetentionMap     map[uuid.UUID]int `json:"-"` // policy for all tasks, injected in runtime
-	RateLimit        []DCLimit         `json:"rate_limit,omitempty"`
-	SnapshotParallel []DCLimit         `json:"snapshot_parallel,omitempty"`
-	UploadParallel   []DCLimit         `json:"upload_parallel,omitempty"`
-	Continue         bool              `json:"continue,omitempty"`
-	PurgeOnly        bool              `json:"purge_only,omitempty"`
+	Units            []Unit     `json:"units,omitempty"`
+	DC               []string   `json:"dc,omitempty"`
+	Location         []Location `json:"location"`
+	Retention        int        `json:"retention"`
+	RetentionDays    int        `json:"retention_days"`
+	RateLimit        []DCLimit  `json:"rate_limit,omitempty"`
+	SnapshotParallel []DCLimit  `json:"snapshot_parallel,omitempty"`
+	UploadParallel   []DCLimit  `json:"upload_parallel,omitempty"`
+	Continue         bool       `json:"continue,omitempty"`
+	PurgeOnly        bool       `json:"purge_only,omitempty"`
+
+	// GetRetention (injected in runtime) returns the policy for all backup
+	// tasks. Returns a default policy if no retention exists.
+	GetRetention RetentionFunc `json:"-"`
```

Comment:
I'd suggest TaskRetention.

Reply:
Moved this out of the target and into the backup service. Now the retention policies are gathered within the backup service when the purge is executed instead of being injected into each target's properties from the scheduler service.

They also no longer need to be exported as they don't to be injected from the integration tests like before.